script.js

```diff
@@ -1,12 +1,273 @@
+           mainDisplayDiv.appendChild(blankColumn);
+    }
+
+}
+
+// EG Winter is Coming - S01E01
+function fetchEpisodeSeason_Full(info) {
+        return `${info.name} - S${String(info.season).padStart(2,"0")}E${String(info.number).padStart(2,"0")}`;
+}
+
+// EG 'S01E01'
+function fetchEpisodeSeason_Suffix(info) {
+        return `S${String(info.season).padStart(2,"0")}E${String(info.number).padStart(2,"0")}`;
+}
+
+
+// Instead of using innerHtml, remove the <p></p> and use innerText instead
+function handle_paragraph(text) {
+      text = text.replace(/<\/?p>/ig,""); // remove any <p> </p>
+      let paragraph = document.createElement("p");
+          paragraph.innerText = text;
+          return paragraph;
+}
+
+// Perform a 'live' search
+function searchFunction(useThisValue) {
+ 
+   searchText = searchBar.value.trim(); // this is the keyed-in value as the user types
+   removeChildren(mainDisplayDiv); // remove previous display
+   if (!searchText) // empty input - show all episodes
+   {
+         showAll(false);
+         return;
+   }
+  
+   let lowerCase = searchText.toLowerCase();
+
+   // filter out matching episodes
+   const searchResults = allEpisodes.filter(element => element.name.toLowerCase().includes(lowerCase) ||
+                                                       element.summary.toLowerCase().includes(lowerCase));
+   
+   if (!searchResults.length) // No match so ignore
+            return;
+
+  for (let i = 0; i < searchResults.length; i++)
+         displaySearchResults(i,searchResults[i],lowerCase);
+  // Display the number of episodes found       
+  displayMessage[0].innerText = `Displaying ${searchResults.length}/${allEpisodesTotal} episodes.`;
+}
+
+// remove HTML nodes
+function removeChildren(node) {
+     while( node.firstChild )
+              node.removeChild( node.firstChild );
+}
+
+function displaySearchResults(index,source,lowerCase) {
+    const episodeDiv = document.createElement('div'); // main 'div' to append to
+
+    let episodeInfo = " - " + fetchEpisodeSeason_Suffix(source); // EG 'S01E01'
+    
+    // change into <span> any occurrence of the search text found within the title
+    // in order to highlight
+    
+    let h1Element = document.createElement('h1');
+    processText(source.name,lowerCase,episodeInfo,h1Element);
+    
+    h1Element.classList.add('episode-title');
+    episodeDiv.appendChild(h1Element);
+
+    const imageDiv = document.createElement('div'); // div for an image
+    imageDiv.classList.add('image');
+    episodeDiv.appendChild(imageDiv);
+
+    const img = document.createElement('img');
+    img.src = source.image.medium;
+    img.alt = episodeInfo;
+    episodeDiv.appendChild(img)
+
+    // change into <span> any occurrence of the search text found within the summary
+    // in order to highlight
+
+    text = source.summary.replace(/<\/?p>/ig,""); // remove any <p> </p>
+    processText(text,lowerCase,"",episodeDiv);
```

Comment:
Is it repeating lines 120-122?

Reply:
No! One is the episode followed by S01E01 etc so Full
The other is simply S01E01 so Suffix